src/responder/kcm/kcmsrv_ops.c

```diff
@@ -1074,6 +1076,73 @@ static void kcm_op_get_principal_getbyname_done(struct tevent_req *subreq)
     tevent_req_done(req);
 }
 
+static void
+kcm_creds_table_delete_cb(hash_entry_t *item,
+                          hash_destroy_enum deltype,
+                          void *pvt)
+{
+    /* Delete the old credential if it is being overwritten. */
+    talloc_free(item->value.ptr);
+}
+
+/* Store credentials in a hash table.
+ *
+ * If the table already exist we add the new credentials to the table and
+ * overwrite the ones that already exist. This allows us to correctly serve
+ * also parallel GET_CRED_UUID_LIST requests from the same connection since
+ * it will have its own uuid list and cursor on the client side and we make
+ * all uuid (old, updated and newly added) available.
+ */
+static errno_t
+kcm_creds_to_table(TALLOC_CTX *mem_ctx,
+                   struct kcm_cred *creds,
+                   hash_table_t **_table)
+{
+    char str[UUID_STR_SIZE];
+    uuid_t uuid;
+    errno_t ret;
+
+    if (*_table == NULL) {
+        *_table = sss_ptr_hash_create(mem_ctx, kcm_creds_table_delete_cb, NULL);
```

Comment:
The worry here is that `sss_ptr_hash_create()` creates table with `count`=10
I suspect this might be **very** inefficient in case we need to store large amount of entries.
See #5134

Reply:
The ticket talks about tens of thousands. Here we talk about hundreds entries, maybe thousands, as very huge ccaches, with tens of entries being common. So even if this parameter was indeed a problem, we should be fine. But we can fine-tune it it indeed turns out to be a problem in this parameter.

From dhash documentation: "The count parameter allows for some more optimization, however it's not critical to get it right, the table will still perform well. If you have no idea of how many entries to expect then pass 0, a reasonable default will be selected.".

Reply:
Ok, fair enough.